### PR TITLE
feat(query-builder): Add basic support for selecting multiple values

### PR DESCRIPTION
**Pull request by malwilley**
Ref https://github.com/getsentry/sentry/issues/69791

For text and number filters, add checkboxes and allow multiple value selection. Other filter types retain the same functionality as before.

Behavior is as follows:

- Clicking the row selects the value and exits the token
- Clicking the checkbox selects the value and keeps focus inside the token
- Pressing backspace will unselect the previous value
- Selected values are always shown at the top of the flyout

https://github.com/getsentry/sentry/assets/10888943/fe541ca6-8a68-477f-889c-f6b42b4e1c5b

### CODE REVIEW

File: static/app/components/searchQueryBuilder/useQueryBuilderState.tsx [hunk 1]
```diff
@@ -101,6 +114,70 @@ function updateFreeText(
...
+        return updateFilterMultipleValues(state, action.token, ['']);
+      }
+      return updateFilterMultipleValues(state, action.token, [
+        tokenValue.text,
+        action.value,
+      ]);
+  }
+}
+
+function deleteLastMutliSelectTokenValue(
```
Comment (MichaelSun48) [inline static/app/components/searchQueryBuilder/useQueryBuilderState.tsx:164]: tiny typo here: Mutli -> Multi

Comment (malwilley) [reply static/app/components/searchQueryBuilder/useQueryBuilderState.tsx:164]: Whoops!

File: static/app/components/searchQueryBuilder/useQueryBuilderState.tsx [hunk 2]
```diff
@@ -101,6 +114,70 @@ function updateFreeText(
...
+
+  const newValue =
+    uniqNonEmptyValues.length > 1
+      ? `[${uniqNonEmptyValues.join(',')}]`
+      : uniqNonEmptyValues[0];
+
+  return {...state, query: replaceQueryToken(state.query, token.value, newValue)};
+}
+
+function mutliSelectTokenValue(
```
Comment (MichaelSun48) [inline static/app/components/searchQueryBuilder/useQueryBuilderState.tsx:137]:
```suggestion
function multiSelectTokenValue(
```
last one I promise
(One usage of this function below too)

File: static/app/components/searchQueryBuilder/useQueryBuilderState.tsx [hunk 3]
```diff
@@ -121,6 +198,10 @@ export function useQueryBuilderState({initialQuery}: {initialQuery: string}) {
           return {
             query: replaceQueryToken(state.query, action.token, action.value),
           };
+        case 'TOGGLE_FILTER_VALUE':
+          return mutliSelectTokenValue(state, action);
```
Comment (MichaelSun48) [inline static/app/components/searchQueryBuilder/useQueryBuilderState.tsx:202]:
```suggestion
          return multiSelectTokenValue(state, action);
```

**Review by MichaelSun48 — APPROVED** [review]
looks amazing! I would definitely get someone else to review too but the UI and the code layout are 👌

File: static/app/components/searchQueryBuilder/useQueryBuilderState.tsx (hunk 2 above)
Comment (malwilley) [reply static/app/components/searchQueryBuilder/useQueryBuilderState.tsx:137]: How do I keep messing this up hah

**Pull request merged**